editor/src/components/canvas/controls/breadcrumb-trail.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react'
+import { useEditorState } from '../../../components/editor/store/store-hook'
+import { ElementPathElement } from '../../../components/inspector/sections/header-section/element-path'
+import {
+  betterReactMemo,
+  useKeepReferenceEqualityIfPossible,
+} from '../../../utils/react-performance'
+import { Utils } from '../../../uuiui-deps'
+import * as TP from '../../../core/shared/template-path'
+import { getOpenUtopiaJSXComponentsFromState } from '../../editor/store/editor-state'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { selectComponents } from '../../../components/editor/actions/action-creators'
+import { FlexRow, Icons } from '../../../uuiui'
+
+export const BreadcrumbTrail = () => {
+  const { dispatch, jsxMetadataKILLME, rootComponents } = useEditorState((store) => {
+    return {
+      dispatch: store.dispatch,
+      jsxMetadataKILLME: store.editor.jsxMetadataKILLME,
+      rootComponents: getOpenUtopiaJSXComponentsFromState(store.editor),
+    }
+  }, 'InspectorContextProvider')
+
+  const onSelect = React.useCallback(
+    (path) => dispatch([selectComponents([path], false)], 'everyone'),
+    [dispatch],
+  )
+
+  const selectedViews = useEditorState(
+    (store) => store.editor.selectedViews,
+    'InspectorEntryPoint selectedViews',
+  )
+
+  const elementPath = useKeepReferenceEqualityIfPossible(
+    React.useMemo(() => {
+      if (selectedViews.length === 0) {
+        return []
+      }
+      let elements: Array<ElementPathElement> = []
+      Utils.fastForEach(TP.allPaths(selectedViews[0]), (path) => {
+        // TODO Scene Implementation
+        if (TP.isInstancePath(path)) {
+          const component = MetadataUtils.getElementByInstancePathMaybe(
+            jsxMetadataKILLME.elements,
+            path,
+          )
+          if (component != null) {
+            elements.push({
+              name: MetadataUtils.getElementLabel(path, jsxMetadataKILLME),
+              path: path,
+            })
+          }
+        } else {
+          const scene = MetadataUtils.findSceneByTemplatePath(jsxMetadataKILLME.components, path)
+          if (scene != null) {
+            elements.push({
+              name: scene.label,
+              path: path,
+            })
+          }
+        }
+      })
+      return elements
+    }, [selectedViews, jsxMetadataKILLME]),
+  )
```

Comment:
I see this is copy pasted code from `inspector.tsx`. Please can you extract it into a hook so that both files use the same code path. The only rule with custom hooks is that they have to have their names must have the `use` prefix, otherwise they're just regular functions.

Reply:
The inspector _I think_ will not need this after this change, because it's only used for the breadcrumbs. @OmarDaSilva did you already remove the now-unused `header-section` etc from the inspector?

Reply:
Ah ok if the other is being binned off then it's fine to leave this as is

Reply:
Commit #8b5b7b1 address this